Dockerfile

```
@@ -0,0 +1,33 @@
+FROM node:lts as build-console
+
+COPY ./console /usr/local/display/console
+WORKDIR /usr/local/display/console
+RUN npm install --no-audit && npm run build
+
+FROM node:lts as build-frontend
+
+COPY ./frontend /usr/local/display/frontend
+WORKDIR /usr/local/display/frontend
+RUN npm install --no-audit && npm run build
+
+FROM node:lts as build-server
+
+COPY ./server /usr/local/display/server
+WORKDIR /usr/local/display/server
+
+RUN npm install --no-audit && npm run compile
+
+FROM node:lts
+COPY --from=build-server /usr/local/display/server/lib /usr/local/display/server
+COPY ./server/public /usr/local/display/server/public
+COPY ./server/package*.json /usr/local/display/server/
+COPY --from=build-console /usr/local/display/console/dist /usr/local/display/server/ext-console
+COPY --from=build-frontend /usr/local/display/frontend/dist /usr/local/display/server/ext-display
+
+WORKDIR /usr/local/display/server
+RUN npm install --only=production --no-audit
+
+EXPOSE 3031
+ENV NODE_ENV production
+ENV NODE_CONFIG_ENV docker
+CMD [ "node", "index.js" ]
```

```suggestion
USER node
CMD [ "node", "index.js" ]
```
The `node:lts` image offers an unprivileged user, so the app does not run as root.